lib/log-parser.js

```diff
@@ -0,0 +1,76 @@
+
+import fs from 'fs'
+import readline from 'readline'
+import LogEntry from 'log-entry'
+import moment from 'moment'
+
+const startDatePattern = /^Started '(.*)' on(.*)$/
+const logEntryPattern = /^\[(\d{2}:\d{2}:\d{2})\][ |\w]:(\t*)(.*)$/
+
+const parseStartDate = (lineWithDate) => {
+  const result = lineWithDate.match(startDatePattern)
+  if (result) {
+    return moment(result[1], 'ddd MMM DD HH:mm:ss PST yyyy').toDate()
+  }
+  return null
+}
+
+const parseLogDate = (logDateString, startDate) => {

```

Comment:
We probably can simplify this part of parsing. Right now we have only time inside log entries like 
```
[02:30:07] : Finalize build settings
```

and we also have full start date in second line of log:
```
Started 'Thu Dec 15 02:37:22 PST 2016' on 'SAC-CORP-TC2' by 'Schedule Trigger
```

so we need to get time from second log line and using time from log entry create full log date for each log entry